Documentation edits to the Getting Started Section (#1510)
eclipse-cyclonedds/cyclonedds: docs/manual/shared_memory/limitations.rst

.. include:: ../external-links.part.rst

.. index:: Shared memory; Limitations

.. _shared_mem_limits:

Limitations of shared memory
============================

> Due to the manner in which the shared memory exchange functions, there are some 

Reviewer: ```suggestion
Due to the manner in which data is exchanged over shared memory, there are some 
```

Author: Updated with suggested change

> limitations to the types of data, and delivery are required to ensure their correct 

Reviewer: ```suggestion
limitations to the types of the data that may be exchanged, and delivery are required to ensure their correct 
```

This is super wonky:
> ..., and delivery are required to ensure their correct ...

the overhead.

.. index:: iceoryx; Data exchange

iceoryx data exchange
---------------------

The manner in which the |url::iceoryx_link| memory pool keeps track of exchanged data puts a 
number of limitations on the QoS settings. For Writers, the following QoS settings 
are prerequisites for shared memory exchange:

.. list-table::
    :align: left
    :widths: 20 80

    * - Liveliness
      - :c:macro:`DDS_LIVELINESS_AUTOMATIC`

    * - Deadline
      - :c:macro:`DDS_INFINITY`

        ``DDS_INFINITY`` is a define in |var-project-short| that indicates an amount 
        of time in excess of the maximum duration that can be represented. It is used 
        to tell the interface that this duration is infinite. If timeouts depend on 
        this duration, they will never occur in the program's lifetime. Due to this, 
        some code paths can differ, and be more efficient.

    * - Reliability
      - :c:macro:`DDS_RELIABILITY_RELIABLE`

    * - Durability
      - :c:macro:`DDS_DURABILITY_VOLATILE`

    * - History
      - :c:macro:`DDS_HISTORY_KEEP_LAST`
        
        The depth is no larger than the publisher history capacity as set in the configuration file

> For Readers, the following QoS settings are prerequisites for shared memory exchange:

Reviewer: ```suggestion
For readers, the following QoS settings are prerequisites for shared memory exchange:
```
